### PR TITLE
drivers: can: mcan: Add cache handling

### PR DESCRIPTION
**Pull request by alexanderwachter**
This PR adds cache handling for the msg_ram.
Some cores that implement m_can, have some D-cache.
Those cores need handling for the cache, because the m_can is connected to the SRAM directly and therefore does not fetch data from the cache.
The result is that data is written to the SRAM in the code, but is only present in the cache.
Register access is not cached, and when the m_can is triggered to operate on the msg_sram, invalid data is processed.
The rule is to invalidate/clear cache after writing to the msg_ram, and before operations are triggered on the m_can.

### CODE REVIEW

**Comment by alexanderwachter** [comment]
Th first commit is part of another PR.

**Comment by henrikbrixandersen** [comment]
@alexanderwachter Please rebase.

File: drivers/can/can_mcan.c [hunk 1]
```diff
@@ -23,6 +23,22 @@ LOG_MODULE_DECLARE(can_driver, CONFIG_CAN_LOG_LEVEL);
 #define MCAN_MAX_DLC CAN_MAX_DLC
 #endif
 
+#if CONFIG_HAS_CMSIS_CORE_M
+#include <arch/arm/aarch32/cortex_m/cmsis.h>
+
+#if __DCACHE_PRESENT == 1
+#define CACHE_INVALIDATE(add, size) SCB_InvalidateDCache_by_Addr((addr), (size))
```
Comment (JeremyBCD) [inline drivers/can/can_mcan.c:30]: This isn't compiling as-is, is the `add` in this macro definition meant to be `addr`?

Comment (alexanderwachter) [reply drivers/can/can_mcan.c:30]: Yep. Sorry, forgot to push again. I tried to compile on top of your PR and actually fixed this locally.

**Review by henrikbrixandersen — APPROVED** [review]

**Review by martinjaeger — APPROVED** [review]

**Pull request merged**